hnn_core/optimization/general_optimization.py

```diff
@@ -0,0 +1,408 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from .metrics import _rmse_evoked
+
+
+class Optimizer:
+    def __init__(self, net, tstop, constraints, set_params, solver='cobyla',
+                 obj_fun='dipole_rmse', scale_factor=1.,
+                 smooth_window_len=None, max_iter=200):
+        """Parameter optimization.
+
+        Parameters
+        ----------
+        net : Network
+            The network object.
+        tstop : float
+            The simulated dipole's duration.
+        constraints : dict
+            The user-defined constraints.
+        set_params : func
+            User-defined function that sets parameters in network drives.
+        solver : string
+            The optimizer, 'bayesian' or 'cobyla'.
+        obj_fun : string
```

Comment:
```suggestion
        obj_fun : str
```

fix also elsewhere ... try to be as consistent as possible with rest of codebase and copy the style. General advice for any open source repo :)